integration/testrunner/db.go

```diff
@@ -16,45 +16,68 @@ package testrunner
 
 import (
 	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
 	"testing"
+	"time"
 
 	"cloud.google.com/go/bigquery"
 	"github.com/sethvargo/go-retry"
+	"google.golang.org/api/iterator"
+	"google.golang.org/genproto/googleapis/cloud/audit"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	logpb "cloud.google.com/go/logging/apiv2/loggingpb"
 )
 
-// queryIfAuditLogExists queries the DB and checks if audit log contained in the query exists or not.
-func queryIfAuditLogExists(ctx context.Context, tb testing.TB, query *bigquery.Query, expectedNum int64) (bool, error) {
+// queryAuditLogs queries the DB and checks if audit log contained in the query exists or not and return the results.
+func queryAuditLogs(ctx context.Context, tb testing.TB, query *bigquery.Query) ([]*logpb.LogEntry, error) {
 	tb.Helper()
-
+	time.Sleep(25 * time.Second)
 	job, err := query.Run(ctx)
 	if err != nil {
-		return false, fmt.Errorf("failed to run query: %w", err)
+		tb.Logf("failed to run query: %s", err.Error())
+		return nil, fmt.Errorf("failed to run query: %w", err)
```

Comment:
I don't know if I understand the value in returning an error and logging the result. Shouldn't we just return the error, then, higher up the stack, log and decide whether we should ignore the error or continue and try again?

Reply:
I forgot to remove these two logs, it was here to help debug a bug previously. I just removed these two loggings.